app.py

```diff
@@ -1,16 +1,91 @@
-from flask import Flask, render_template
-
+from flask import Flask, render_template, make_response, g
+import os
+from typing import final
+import pymysql
+import psycopg2
+from urllib.parse import unquote
+import traceback
+from flask_restful import reqparse
 from tasks.banner import BannerTask
 
 app = Flask(__name__)
 
+password_decoded = unquote(os.environ.get('DB_PASSWORD'))
 
 @app.route("/")
 def main():
     return "Quay Service Tool Backend"
 
 app.add_resource(BannerTask, '/banner')
 
+@app.before_request
+def before_request():
+    g.db = psycopg2.connect(host=os.environ.get('DB_HOST'), port = os.environ.get('DB_PORT'), database=os.environ.get('DB_NAME'), user=os.environ.get('DB_USER'), password=password_decoded)
+
+@app.teardown_request
+def teardown_request(exception):
+    g.db.close()
+    
+@app.route("/update_banner", methods=['PUT'])
```

Comment:
consider using a RESTUL API interface and moving each "task" to its own file. Checkout https://github.com/quay/quay-service-tool/blob/main/tasks/banner.py where I have a basic example of this using `flask_restful`